packages/components/src/Steps.js

```diff
@@ -1,10 +1,30 @@
 import React from 'react'
 import { useDeck } from './context'
 import useSteps from './useSteps'
+import { Step } from './useStep'
 
-export const Steps = props => {
-  const step = useSteps(props.length)
-  return props.render({ step })
+export const Steps = ({ items, render, ...props }) => {
+  const step = useSteps(items ? items.length : props.length)
+  return items ? (
+    <div>
+      {items.map((item, i) => {
+        const visible = props.split ? step === i : step >= i
+        return (
+          visible && (
+            <Step key={i} index={i}>
+              {render
+                ? render(item, i)
+                : typeof item === 'function'
+                ? item(i)
+                : item}
```

Comment:
When the `items` prop exists, the `render` prop can be used to map the items into elements. Alternatively, you can pass a function in the `items` array prop, and it will be passed its index in the array.